runtime/environment.go

```diff
@@ -355,103 +355,42 @@ func (e *interpreterEnvironment) ParseAndCheckProgram(
 func (e *interpreterEnvironment) parseAndCheckProgram(
```

Comment:
`parseAndCheckProgram` no longer has the responsibility of calling `SetProgram` if `storeProgram` is true